### DISCUSSION TITLE
Specifying Client IP

### DISCUSSION THREAD

**Discussion by LazyHippogriff** [uNetworking/uWebSockets#1314] [Q&A]
How can I specify which IP should be used by the client when connecting to a server? I've not been able to find any option till now. Please help! I'm using v0.14.

**Comment by ghost** [comment]
That's probably not supported.

**Reply by LazyHippogriff** [reply]
Do you know if there is any deterministic logic by which client IP is chosen? Or is it random?

**Reply by ghost** [reply]
It's called bind-before-connect. You call bind on the socket before you call connect. That's how you can specify on what interface(IP) to connect using. That's not supported in v0.14, I don't think so. You can probably PR it or do it in your own fork if you need to.

**Reply by ddevienne** [reply]
https://idea.popcount.org/2014-04-03-bind-before-connect/

Thanks for that name. Allowed to find that interesting article.
